### PR TITLE
Add .clang-format file

### PR DESCRIPTION
**Pull request by ffoulkes**

### CODE REVIEW

**Comment by 5abeel** [comment]
Does this require clang-format installed in dev environment?

**Comment by ffoulkes** [comment]
> Does this require clang-format installed in dev environment?

No, it doesn't. It specifies the defaults to use if you run clang-format without explicit options. This particular file was copied from stratum.

**Review by 5abeel — APPROVED** [review]
LGTM

**Comment by nupuruttarwar** [comment]
@ffoulkes I remember our conversation about clang format as a recommended for stratum codebase. Should we make it as part of some guideline?

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Comment by 5abeel** [comment]
> @ffoulkes I remember our conversation about clang format as a recommended for stratum codebase. Should we make it as part of some guideline?

Might be good to do that (if you have the time, based on priority list)

**Comment by nupuruttarwar** [comment]
> > @ffoulkes I remember our conversation about clang format as a recommended for stratum codebase. Should we make it as part of some guideline?
> 
> Might be good to do that (if you have the time, based on priority list)

Sure, we can take this as part of documentation update.

**Comment by ffoulkes** [comment]
> @ffoulkes I remember our conversation about clang format as a recommended for stratum codebase. Should we make it as part of some guideline?

Yes, we should.

As I see it, the process would be something along the lines of:
- Develop a guideline
- Commit `.clang-format` files where appropriate
- Develop a GitHub action to run clang-format in check mode
- Pick an appropriate moment to put the policy into effect
- Let people know that it's going into effect
- Reformat and push updates in batches
- Enable the GitHub checker
- Advise developers to
  - Update their working repositories
  - Run clang-format before pushing updates

Basically, we want the initial commit to consist _only_ of format changes.

A developer can:
- Run clang-format on their work in progress
- Stash their work
- Pull a fresh copy of the repository
- Pop or apply the stashed copy
- Deal with merge errors and commit

If you want to take out some insurance, you can stash your WIP, apply it, clang-format it, and stash again. Then pull fresh copies of the repos. The second stash (the one on the top of the stack) should merge fairly cleanly. If something goes wrong, you've still got your original WIP to fall back on.
